pkg/cloud/awsprovider.go

```diff
@@ -370,6 +391,23 @@ type AwsAthenaInfo struct {
 	MasterPayerARN   string `json:"masterPayerARN"`
 }
 
+// CreateConfig creates an AWS SDK V2 Config for the credentials that it contains
+func (aai *AwsAthenaInfo) CreateConfig() (awsSDK.Config, error) {
+	keyProvider := AWSAccessKey{AccessKeyID: aai.ServiceKeyName, SecretAccessKey: aai.ServiceKeySecret}
+	cfg, err := keyProvider.CreateConfig(aai.AthenaRegion)
+	if err != nil {
+		return cfg, err
+	}
+	if aai.MasterPayerARN != "" {
+		// Create the credentials from AssumeRoleProvider to assume the role
+		// referenced by the roleARN.
+		stsSvc := sts.NewFromConfig(cfg)
+		creds := stscreds.NewAssumeRoleProvider(stsSvc, aai.MasterPayerARN)
+		cfg.Credentials = awsSDK.NewCredentialsCache(creds)
+	}
+	return cfg, nil
+}
+
```

Comment:
Configuration method for Athena SDK v2, locks region to athenaRegion and also applies Master Payer ARN if applicable.